### PR TITLE
refactor: construct ics27 error acknowledgement with determinstic ABCI code

### PR DESCRIPTION
**Pull request by colin-axner**
## Description

Splitting #701 into 3 parts:
- error acknowledgement changes
- results acknowledgement changes
- ADR explaining justification

This pr handles the first case

ref: #701 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [x] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [x] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [x] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: modules/apps/27-interchain-accounts/host/types/ack_test.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	suite.Run(t, new(TypesTestSuite))
+}
+
+// The safety of including ABCI error codes in the acknowledgement rests
+// on the inclusion of these ABCI error codes in the abcitypes.ResposneDeliverTx
+// hash. If the ABCI codes get removed from consensus they must no longer be used
+// in the packet acknowledgement.
+//
+// This test acts as an indicator that the ABCI error codes may no longer be deterministic.
+func (suite *TypesTestSuite) TestABCICodeDeterminism() {
```
Comment (damiannolan) [inline modules/apps/27-interchain-accounts/host/types/ack_test.go:47]: Nice! This is super useful :)

**Review by damiannolan — APPROVED** [review]

**Review by seantking — APPROVED** [review]

File: modules/apps/27-interchain-accounts/host/types/ack_test.go (hunk 1 above)
Comment (seantking) [reply modules/apps/27-interchain-accounts/host/types/ack_test.go:47]: yeah, this is cool!

**Review by crodriguezvega — COMMENTED** [review]
👮 Spelling police! 🚔 There is a typo (`determinstic`) in the comments in the `modules/apps/27-interchain-accounts/host/types/ack.go` file. 😸

File: modules/apps/27-interchain-accounts/host/types/ack.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	channeltypes "github.com/cosmos/ibc-go/v3/modules/core/04-channel/types"
+)
+
+const (
+	// ackErrorString defines a string constant included in error acknowledgements
+	// NOTE: Changing this const is state machine breaking as acknowledgements are written into state
+	ackErrorString = "error handling packet on host chain: see events for details"
+)
+
+// AcknowledgementErrorString returns a determinstic error string which may be used in
```
Comment (colin-axner) [inline modules/apps/27-interchain-accounts/host/types/ack.go:17]:
```suggestion
// AcknowledgementErrorString returns a deterministic error string which may be used in
```

**Pull request merged**